Change default poll frequency when idle form 1000hz to 10hz.

baldersheim opened the pull request:
The cost of a poll is huge as it implies sending an http post.
Synchrous usage might want to reduce this.

@bratseth or @bjorncs PR
This reverts the effective change in #9954, but makes it configurable.
Sync usage should use a sync protocol instead.

baldersheim committed:
Change default poll frequency when idle form 1000hz to 10hz.
The cost of a poll is huge as it implies sending an http post.
Synchrous usage might want to reduce this.

bratseth reviewed (commented):
> Sync usage should use a sync protocol instead.

It's not that simple since there is one code path, which some times happen to get many updates, and some times a few for a given poll to the system delivering updates. 

This will bring back the problem fixed by the previous PR ... unless people know which lever to pull to fix it, but that's very hard. Could we think about whether this could be done in a smarter way, how about keeping track of how long it was since we polled?

baldersheim committed:
- Let default idlePollFrequency be unset.
- If unset SyncClient will set it to 200hz.
- If not set IOThread will use old default of 10hz.

baldersheim committed:
Let fast be fast

baldersheim commented:
@bratseth Now you can have a look again.
Another option would be to use exponential backoff in the range [1-100], Fx 1, 2, 4 in the sync case.

baldersheim committed:
1000 -> 500.0

bratseth reviewed (approved):
💯
